src/core/exec-invoke.c

```diff
@@ -5423,6 +5462,36 @@ int exec_invoke(
         if (r < 0)
                 return r;
 
+        if (context->private_bpf) {
+                _cleanup_close_ int fs_fd = -EBADF, mnt_fd = -EBADF, bpffs_fd = -EBADF;
+
+                fs_fd = fsopen("bpf", 0);
+                if (fs_fd < 0)
+                        return fs_fd;
+
+                r = send_one_fd(token_fds[0], fs_fd, 0);
+                if (r < 0)
+                        return r;
+
+                mnt_fd = receive_one_fd(token_fds[0], 0);
+                if (mnt_fd < 0)
+                        return fs_fd;
+
+                bpffs_fd = openat(mnt_fd, ".", O_RDONLY);
+                if (bpffs_fd < 0)
+                        return bpffs_fd;
```

Comment:
Why is it necessary to open??

Reply:
It was present in the sample code, but maybe it's unneeded now that I'm not creating the token anymore:

https://elixir.bootlin.com/linux/v6.14.6/source/tools/testing/selftests/bpf/prog_tests/token.c#L374